src/qengine/qengine.cpp

```diff
@@ -558,4 +558,130 @@ bitCapInt QEngine::ForceMReg(bitLenInt start, bitLenInt length, bitCapInt result
     return result;
 }
 
+/// Add integer (without sign, with carry)
+void QEngine::INCC(bitCapInt toAdd, const bitLenInt inOutStart, const bitLenInt length, const bitLenInt carryIndex)
```

Comment:
All of these math functions really seem like they should live in a `QMath` baseclass to keep the core quantum operator routines easy to find and read - and also to break apart these files (which are getting to the unwieldy point).

Reply:
Notice that there are already almost exclusively arithmetic methods in the file. I renamed this file "arithmetic.cpp" and separated the couple of other methods into appropriate bins. I also separated "logic.cpp" from "arithmetic.cpp" throughout. (I had to add new file headers with new copyright dates, so I also updated the copyright notices throughout the library.)